File: javascript/js/vitalsigns.js
import Client from "./client";

class VitalSignsClient {
   async get() {
    const client = new Client();
    return client.request("Observation/1");
  }

  async search() {
    const client = new Client();
    return client.request("Observation?identifier=1234&device-name=testDevice");
  }
}

export default VitalSignsClient;

File: ruby/src/vitalsigns.rb
require './fhir_client'

class VitalSignsClient
  def get_vitalsigns(id)
    FhirClient.new

    FHIR::Observation.read(id)
  end

  def search_vitalsigns
    FhirClient.new

    params = {
      'identifier': '1234',
      'device-name': 'testDevice'
    }
    FHIR::Observation.search(params)
  end
end
